install_script_op_worker2.sh

```diff
@@ -475,6 +533,64 @@ else
     printf "\033[34m  * Assigning DD_SITE.\n\033[0m\n"
     $sudo_cmd sh -c "echo DD_SITE=$site >> $env_file"
   fi
+
+  # sources
+  if [ "$source_splunk_hec_address" ]; then
+    printf "\033[34m  * Assigning DD_OP_SOURCE_SPLUNK_HEC_ADDRESS.\n\033[0m\n"
+    $sudo_cmd sh -c "echo DD_OP_SOURCE_SPLUNK_HEC_ADDRESS=$source_splunk_hec_address>> $env_file"
+  fi
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏I think this is just my brain not quite working fully - but with `source_splunk_hec_address=`, meaning the environment variable isn't set, this conditional fails and we don't write it to the file?

Reply:
Correct, we dont write to the $env_file for the respective env var, if the the relevant DD_OP_* env var is not set

From the customer's use case, they will be presented with a snippet to copy-paste and run. And the relevant env vars they need based on their pipeline are prepended to the install script invocation.

Reply:
👍 for a second I couldn't remember if `[ "$foo" ]` was true if the variable was set at all, and if "empty" counted as set. Bash is fun sometimes 😉

Reply:
This should probably use `-n` to test if the variable is not empty, avoiding another trap of variables values starting with a hyphen causing confusion.